app/src/main/java/org/p2p/wallet/newsend/ui/NewSendPresenter.kt

```diff
@@ -153,6 +155,14 @@ class NewSendPresenter(
             }
 
             initializeFeeRelayer(view, initialToken, solToken)
+            initialAmount?.let { inputAmount ->
+                if (calculationMode.getCurrencyMode() is CurrencyMode.Fiat.Usd) {
+                    switchCurrencyMode()
+                }
+                view.updateInputValue(inputAmount.scaleShort().toPlainString(), forced = true)
+                calculationMode.updateInputAmount(inputAmount.scaleShort().toPlainString())
+                view.disableInputs()
+            }
```

Comment:
could you please move to a separate function? 
Something like `setupDefaultFields`